build-tools/packages/build-cli/src/handlers/checkFunctions.ts

```diff
@@ -903,3 +903,54 @@ export const checkValidReleaseGroup: StateHandlerFunction = async (
 
 	return true;
 };
+
+/**
+ * Checks whether layer compat generation needs to be updated.
+ *
+ * @param state - The current state machine state.
+ * @param machine - The state machine.
+ * @param testMode - Set to true to run function in test mode.
+ * @param log - A logger that the function can use for logging.
+ * @param data - An object with handler-specific contextual data.
+ * @returns True if the state was handled; false otherwise.
+ */
+export const checkLayerCompatGeneration: StateHandlerFunction = async (
```

Comment:
The terminology still really confuses me here. We have a "layer check" tool. Are these the same layers?